### DISCUSSION TITLE
After reinstalling the system, whenever I try to use Subversion after installing it, I keep getting errors E170013 and E120171. Unable to operate properly.

### DISCUSSION THREAD

**Discussion by helee9199** [Homebrew/discussions#4911] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: d3d51d3448966b5250b5c60110c945040054803d
Last commit: 5 days ago
Core tap JSON: 17 Nov 01:08 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_API_DOMAIN: https://mirrors.tuna.tsinghua.edu.cn/homebrew-bottles/api
HOMEBREW_BOTTLE_DOMAIN: https://mirrors.ustc.edu.cn/homebrew-bottles
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_PIP_INDEX_URL: https://pypi.tuna.tsinghua.edu.cn/simple
all_proxy: socks5://127.0.0.1:7890
http_proxy: http://127.0.0.1:7890
https_proxy: http://127.0.0.1:7890
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 12.7.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I recently reinstalled my computer's operating system. After installing SVN via Homebrew, when attempting to use SVN operations, I received the following error message:
svn: E170013: Unable to connect to a repository at URL 'https://xxxxxxx’
svn: E120171: Error running context: An error occurred during SSL communication
I searched extensively across the internet for solutions and found two similar situations with suggested fixes:
https://superuser.com/questions/1473219/subversion-error-svn-e120171-error-running-context-an-error-occurred-during
https://stackoverflow.com/questions/72201289/cant-do-svn-checkout-because-of-tls-error

However, after following the adjustments mentioned in those solutions, the issue remains unresolved. I even attempted reinstalling macOS 12 and macOS 14, but encountered the same problem. Additionally, a friend who recently bought a new MacBook Pro also experienced the same issue after installing SVN.

At my workplace, I have a Windows computer where SVN version 1.14.2 is installed. In the same network environment, SVN works on Windows, but not on macOS. I used to have no problems with SVN usage until December 2021, just after purchasing the computer. However, now, after using the computer for two years and attempting a system reinstallation, this issue has surfaced.

I am at my wit's end. I spent the entire day yesterday trying to resolve this problem, hoping someone could provide assistance.

### Relevant casks

subversion 1.14.2_4

**Comment by SMillerDev** [comment]
That sounds like a question you should ask the SVN devs. Additionally, it could very well be an issue with your firewall.

**Reply by helee9199** [reply]
But the server configuration has not changed for so many years. Because the svnserve is still the 1.6.6 version deployed in 2009, it is TLSv1 and there is no way to change it. The company also does not plan to upgrade. Maybe the new version installed by brew is not supported. Is there a way to install the version around the beginning of 2021? It was still usable when I installed it at that time.

**Comment by Jerry1144** [comment]
I believe you can `git checkout -C $(brew --repo homebrew/core) [insert old commit here] -- Formula/subversion.rb`, move it to `"$(brew --repo homebrew/core)"/Formula/s/subversion.rb`, and then `HOMEBREW_NO_AUTO_UPDATE=1 brew install subversion` and such. _If you really think an old svn is the solution._

Though it may well be a certificate problem, along with insecure protocol explicitly disabled. Someone well-versed in openssl may tell you how to make command-line tools fetch the System Keychain certificates after-the-fact. Or first of all, a `openssl -v ...` one-liner that tells you exactly what connection error it is.
```
brew info openssl@3
[snip]
==> Caveats
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /usr/local/etc/openssl@3/certs

and run
  /usr/local/opt/openssl@3/bin/c_rehash
```
Looking into the formula, it seems not to be bootstrapped from System Keychain, but instead using Mozilla's CA store…

**Reply by Jerry1144** [reply]
Sorry it's `git -C "$(brew --repo homebrew/core)" checkout [commit hash] -- Formula/subversion.rb`. The -C option has to come before the subcommand to have the intended effect.
